pkg/controller/machinepool/machinepool_controller.go

```diff
@@ -582,20 +582,36 @@ func (r *ReconcileMachinePool) syncMachineSets(
 					}
 				}
 
-				// Update if the labels on the remote machineset are different than the labels on the generated machineset.
-				// If the length of both labels is zero, then they match, even if one is a nil map and the other is an empty map.
-				if rl, l := rMS.Spec.Template.Spec.Labels, ms.Spec.Template.Spec.Labels; (len(rl) != 0 || len(l) != 0) && !reflect.DeepEqual(rl, l) {
-					msLog.WithField("desired", l).WithField("observed", rl).Info("labels out of sync")
```

Comment:
I had to get rid of this log line because I removed the deepequal comparison, but if an update is called - there would be a log entry for that.